python/cudf/cudf/core/index.py

```diff
@@ -2364,7 +2364,12 @@ def isocalendar(self):
 
     @_cudf_nvtx_annotate
     def to_pandas(self, nullable=False):
-        nanos = self._values.astype("datetime64[ns]")
+        if isinstance(self.dtype, pd.DatetimeTZDtype):
+            nanos = self._values.astype(
+                pd.DatetimeTZDtype("ns", self.dtype.tz)
```

Comment:
As a pandas 2.0 note, would be good to keep "s", "ms", or "us" resolution here

Reply:
Since this is currently targeting `branch-23.06` which doesn't have pandas-2.0 support this change seems okay, what @mroeschke suggesting is taken care of in `pandas_2.0_feature_branch`, so perhaps a note or todo here would be useful while I sync these changes with pandas-2.0 support.